README.md

```diff
@@ -15,3 +15,49 @@ To install ML Metadata, run:
     juju deploy mlmd
 
 For more information, see https://juju.is/docs
+
+## Upgrade
+
+This action can be performed with:
+
+```
+juju refresh mlmd --channel <desired-channel>
+```
+
+#### Upgrading from `mlmd<=1.14/stable`
+
+> WARNING: to correctly perform this migration, you must [backup your data](https://discourse.charmhub.io/t/data-backups-and-restoration-for-ckf/13999#heading--backup-mlmd).
+
+1. Remove the relation with requirer charms (e.g. `envoy` and `kfp-metadata-writer`)
+
+```
+juju remove-relation envoy mlmd
+juju remove-relation kfp-metadata-writer mlmd
+```
+
+2. Remove the `mlmd` application.
+
+> WARNING: this will wipe out the storagae attached to the `mlmd` charm, therefore, the database
+that this charm handles. It is important you perform a [data backup](https://discourse.charmhub.io/t/data-backups-and-restoration-for-ckf/13999#heading--backup-mlmd) before
+running this step.
+
+```
+juju remove-application mlmd --destroy-storage
+```
+
+3. Deploy `mlmd` from a newer channel
+
+```
+juju deploy mlmd --channel <channel-greater-than-1.14/stable> --trust
+```
+
+4. [Restore your data](https://discourse.charmhub.io/t/data-backups-and-restoration-for-ckf/13999#heading--restore-mlmd).
+
+5. Relate to the requirer charms
+
+> WARNING: this relation is only supported for `envoy>2.0/stable` and `kfp-metadata-writer>2.0/stable`
```

Comment:
Maybe rephrase to sth like: `mlmd>1.14/stable` supports the following relations only for `envoy>2.0/stable` and `kfp-metadata-writer>2.0/stable`